removed necessity of nmap

RobinBol opened the pull request:
Replaced functionality provided by node-nmap, by using a regular TCP
scan. As node-nmap requires a binary (nmap), which can not be used on
the platform we deploy ikettle.js on, overall it is better to use modules that do not require binaries to increase cross-platform support.

RobinBol committed:
removed necessity of nmap

Replaced functionality provided by node-nmap, by using a regular TCP
scan. As node-nmap requires a binary (map), which can not be used on
the platform we deploy ikettle.js on.

lloydwatkin commented:
Hey, thanks for the PR. Would you be able to update the related tests/add new tests/remove irrelevant ones. Once that's in place I'd be happy to get this in?

Out of interest where are you deploying this module?

I'm getting fed up of the android app not being able to discover my kettle after a while so am in the process of creating a home application/site which will allow me to control from there (including other things like lights).

RobinBol committed:
Updated discover test

Removed redundant tests and added fitting ones for the tcp port
scanning support

RobinBol committed:
fixed proxyquire mistake

RobinBol committed:
discover parameters -> kettle

RobinBol committed:
Added proper working tests

RobinBol committed:
callback called multiple times

RobinBol commented:
I removed irrelevant tests and added proper new ones. Testing is rather new for me, so bear with me if it is not what you would like/expect. About multi-device support, for now I have no time to work on that, but that would be great! Only problem I foresee is that you can't label your kettle, as they don't come with a name or something. This could of course be fixed by just iteratively naming them Kettle 1 and Kettle 2, but this is not very user friendly, as user would like to know what kettle (in what room in the house) they are interacting with. Any ideas on that?

lloydwatkin commented:
MAC address mapping?

Will get this published this evening, thanks!

lloydwatkin commented:
Published as 2.0.0. Thanks
